src/main/scala/is/hail/expr/ir/ArraySorter.scala

```diff
@@ -8,27 +8,37 @@ import is.hail.utils._
 class ArraySorter(mb: EmitMethodBuilder, array: StagedArrayBuilder, keyOnly: Boolean) {
   val typ: Type = array.elt
   val ti: TypeInfo[_] = typeToTypeInfo(typ)
-  val sortmb: EmitMethodBuilder = mb.fb.newMethod[Region, Int, Int, Unit]
+  val sortmb: EmitMethodBuilder = mb.fb.newMethod[Region, Int, Int, Boolean, Unit]
 
-  val (compare: CodeOrdering.F[Int], equiv: CodeOrdering.F[Boolean]) = if (keyOnly) {
+  val (less: CodeOrdering.F[Boolean], greater: CodeOrdering.F[Boolean], equiv: CodeOrdering.F[Boolean]) = if (keyOnly) {
     val ttype = coerce[TBaseStruct](typ)
     require(ttype.size == 2)
     val kt = ttype.types(0)
+    val mk1s = sortmb.newLocal[Boolean]
```

Comment:
Does it matter whether these variables are defined individually in `clt` and `cgt` or outside of the closures?

Reply:
They shouldn't be inside the closures, or it would add a new local variable each time the closure is called. They could go inside the `clt` and `cgt` definitions before the closures, if you think that's clearer.